typed/parser.go

```diff
@@ -69,79 +69,52 @@ func (p *Parser) TypeNames() (names []string) {
 // Type returns a helper which can produce objects of the given type. Any
 // errors are deferred until a further function is called.
 func (p *Parser) Type(name string) ParseableType {
-	return &parseableType{
-		parser:   p,
-		typename: name,
+	return ParseableType{
+		Schema:  &p.Schema,
+		TypeRef: schema.TypeRef{NamedType: &name},
 	}
 }
 
 // ParseableType allows for easy production of typed objects.
-type ParseableType interface {
-	IsValid() bool
-	FromYAML(YAMLObject) (TypedValue, error)
-	FromUnstructured(interface{}) (TypedValue, error)
+type ParseableType struct {
+	schema.TypeRef
```

Comment:
I'm not a huge fan of the embedding ... (I think this is composition)

Reply:
Fixed